calico_node/ip_detection.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+)
+
+func detectIPs(ignoreInterfaces []string) ([]string, error) {
+
+	ips := []string{}
+	ifaces, err := net.Interfaces()
+
+	for _, iface := range ifaces {
+
+		valid_ip := true
+
+		// Check each interface name against 
+		// list of interfaces to ignore.
+		for _, v := range ignoreInterfaces {
+			regex, err := regexp.Compile(v)
+
+			if err != nil {
+				continue
```

Comment:
There's no indication of an error when we hit this - we should at least be logging something urgent here, and at most be panicking.